app/assets/javascripts/repositories/stock.js

```diff
@@ -18,7 +18,7 @@ var RepositoryStockValues = (function() {
             closeOnSelect: true,
             selectAppearance: 'simple',
             onChange: function() {
-              $('.stock-final-container .units').text(dropdownSelector.getValues(UNIT_SELECTOR));
+              $('.stock-final-container .units').text(dropdownSelector.getData(UNIT_SELECTOR)[0].label.trim());
```

Comment:
Maybe you can add `getLabels` method to the dropdown? It must be a copy of getValues just a different field?